crowdfunding/users/serializers.py

```diff
@@ -0,0 +1,73 @@
+from projects.serializers import PledgeSerializer, ProjectSerializer
+from rest_framework import serializers, validators
+from .models import CustomUser
+
+# class CustomUserSerializer(serializers.Serializer):
+    # id = serializers.ReadOnlyField()
+    # username = serializers.CharField(max_length=150)
+    # email = serializers.EmailField()
+# 
+    # def create(self, validated_data):
+      	# return CustomUser.objects.create(**validated_data) #validated by serializer??
+# 
+# class CustomUserSerializer(serializers.ModelSerializer):
+# 
+    # class Meta:
+        # model = CustomUser
+        # fields = ('email', 'username', 'password', 'avatar', 'bio')
+        # extra_kwargs = {'password': {'write_only': True}}
+    # 
+    # def create(self, validated_data):
+        # user = CustomUser(
+            # email=validated_data['email'],
+            # username=validated_data['username']
+        # )
+        # user.set_password(validated_data['password'])
+        # user.save()
+        # return user
+        
+class CustomUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CustomUser
+        fields = ("id", "username", "email", "is_active", "password", "bio", "avatar")
+
+        extra_kwargs = {
+            "email": {
+                "validators": [validators.UniqueValidator(queryset=CustomUser.objects.all())],
```

Comment:
i would have thought this was pretty much built in behavior for a model serializer? 
did this need to be provided?